GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -1537,6 +1506,89 @@ static IEnumerable<ObjectId> TryGetParents(IGitModule module, FilterInfo filterI
             }
         }
 
+        /// <summary>
+        /// Returns the name of a file in a specific revision (following renames and merge commits).
+        /// </summary>
+        /// <param name="path">The path to the file to get the name of</param>
+        /// <param name="objectId">The revision to get the file name in</param>
+        /// <returns>The name of the file at <paramref name="path"/> in revision identified by <paramref name="objectId"/>; <see langword="null"/> if not available.</returns>
+        public string? GetRevisionFileName(string path, ObjectId? objectId)
+        {
+            if (objectId is null)
+            {
+                return null;
+            }
+
+            if (FilePathByObjectId?.TryGetValue(objectId, out string? fileName) is true)
+            {
+                return fileName;
+            }
+
+            const string startOfObjectId = "????";
+
+            GitArgumentBuilder args = new("log")
+            {
+                // --name-only will list each filename on a separate line, ending with an empty line
+                // Find start of a new commit with a sequence impossible in a filename
+                $"--format=\"{startOfObjectId}%H\"",
+                "--name-only",
+                "--follow",
+                "--diff-merges=separate",
+                FindRenamesAndCopiesOpts(),
+                $"{objectId}^1..{objectId}",
```

Comment:
Might the following be more efficient?
```suggestion
                $"{objectId}",
                "--max-count=1"
```